File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        
        i = 0 
        j = k-1
        c = 0
        sumi = sum(arr[:k-1])
        while j<len(arr):
            sumi = sumi+arr[i+k-1]
            avg = sumi/k
            if avg>=threshold:
                c+=1
            sumi -= arr[i]
            i+=1
            j+=1
        return c
            
            

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        
        int c = 0;
        double sum = 0;
        double avg = 0;

        for(int i=0;i<k;i++){
            sum+=arr[i];
        }
        avg = sum/k;
        if(avg>=threshold){
            c++;
        }
        for(int j=k;j<arr.length;j++){
            sum -= arr[j-k];
            sum += arr[j];
            avg = sum/k;
            if(avg>=threshold){
                c++;
            }
        }

        return c;






    }
}